forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/LocalDeviceRepository.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2022 - 2023 ForgeRock. All rights reserved.
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license. See the LICENSE file for details.
+ */
+
+package org.forgerock.android.auth.devicebind
+
+import android.content.Context
+import android.content.SharedPreferences
+import org.forgerock.android.auth.EncryptedPreferences
+import org.forgerock.android.auth.callback.DeviceBindingAuthenticationType
+import org.forgerock.android.auth.exception.ApiException
+import org.json.JSONObject
+import java.io.IOException
+
+const val ORG_FORGEROCK_V_1_DEVICE_REPO = "org.forgerock.v1.DEVICE_REPO"
+
+interface DeviceRepository {
+
+    suspend fun persist(userKey: UserKey)
+
+    suspend fun getAllKeys(): List<UserKey>
+
+    @Throws(ApiException::class, IOException::class)
+    suspend fun delete(userKey: UserKey)
+
+}
+
+const val idKey = "id"
+const val userIdKey = "userId"
+const val kidKey = "kid"
+const val authTypeKey = "authType"
+const val userNameKey = "username"
+
+/**
+ * Helper class to save and retrieve EncryptedMessage
+ */
+internal class LocalDeviceRepository(context: Context,
```

Comment:
Rename to LocalDeviceBindingRepository ...

Reply:
Updated